src/manifolds/SymmetricPositiveDefinite.jl

```diff
@@ -30,10 +30,100 @@ struct SymmetricPositiveDefinite{N} <: AbstractDecoratorManifold{ℝ} end
 
 SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
 
+@doc raw"""
+    SPDPoint <: AbstractManifoldsPoint
+
+Store the result of `eigen(p)` of an SPD matrix and (optionally) ``p^{\frac{1}[2}}`` and ``p^{-\frac{1}[2}}``
+to avoid their repeated computations.
+
+This result only has the result of `eigen` as a mandatory storage, the other three
+can be stored. If they are not stored they are computed and returned (but then still not stored)
+when required.
+
+# Constructor
+
+    SPDPoint(p::AbstractMatrix; store_p=true, store_sqrt=true, store_sqrt_inv=true)
+
+Create an SPD point using an symmetric positive defincite matrix `p`, where you can optionally store `p`, `p_sqrt` and `p_sqrt_inv`
+"""
+struct SPDPoint{
+    P<:Union{AbstractMatrix,Missing},
+    Q<:Union{AbstractMatrix,Missing},
+    R<:Union{AbstractMatrix,Missing},
+    E<:Eigen,
+} <: AbstractManifoldPoint
+    p::P
+    eigen::E
+    sqrt::Q
+    sqrt_inv::R
+end
+SPDPoint(p::SPDPoint) = p
+function SPDPoint(p::AbstractMatrix; store_p=true, store_sqrt=true, store_sqrt_inv=true)
+    e = eigen(Symmetric(p))
+    U = e.vectors
+    S = max.(e.values, floatmin(eltype(e.values)))
+    if store_sqrt
+        s_sqrt = Diagonal(sqrt.(S))
+        p_sqrt = U * s_sqrt * transpose(U)
+    else
+        p_sqrt = missing
+    end
+    if store_sqrt_inv
+        s_sqrt_inv = Diagonal(1 ./ sqrt.(S))
+        p_sqrt_inv = U * s_sqrt_inv * transpose(U)
+    else
+        p_sqrt_inv = missing
+    end
+    if store_p
+        q = p
+    else
+        q = missing
+    end
+    return SPDPoint{typeof(q),typeof(p_sqrt),typeof(p_sqrt_inv),typeof(e)}(
+        q,
+        e,
+        p_sqrt,
+        p_sqrt_inv,
+    )
+end
+
+convert(::Type{SPDPoint}, p::AbstractMatrix) = SPDPoint(p)
```

Comment:
I think I defined a `Matrix` variant here, right? My idea was (similar to `I` that it would “materialize” the matrix into its array form – and you now use a convert (a different one from this) to do that. What is the advantage?

Reply:
The advantage is that `convert` that works on an `AbstractMatrix` doesn't do anything, while `Matrix` constructor would make a copy, and furthermore be problematic when working with other array types like `SMatrix`.

Reply:
ok, thanks for the explanation.